CMakeLists.txt

```diff
@@ -447,7 +453,10 @@ include(CLangFormat)
 
 set(QT_COMPONENTS Core Network Concurrent Gui Svg Widgets Test LinguistTools)
 if(UNIX AND NOT APPLE)
-    find_package(Qt5 COMPONENTS ${QT_COMPONENTS} DBus X11Extras REQUIRED)
+    if(WITH_XC_X11)
+        set(QT_COMPONENTS ${QT_COMPONENTS} X11Extras)
```

Comment:
I prefer to use `list(APPEND QT_COMPONENTS X11Extras)`, it makes the action clearer. One of the many refactors we need to do in our CMakeLists in general.